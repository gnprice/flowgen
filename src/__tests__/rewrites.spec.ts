import { compiler, beautify } from "..";
import "../test-matchers";

test("rewrite idiosyncrasies of the react-native TS definitions", () => {
  const ts = `
import * as ReactNative from 'react-native';
type A = ReactNative.Constructor<number>;
`;
  const result = compiler.compileDefinitionString(ts, { quiet: true });
  expect(beautify(result)).toMatchSnapshot();
  //   expect(result).toBeValidFlowTypeDeclarations();
});
