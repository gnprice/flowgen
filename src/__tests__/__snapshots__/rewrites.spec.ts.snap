// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rewrite idiosyncrasies of the react-native TS definitions 1`] = `
"import * as ReactNative from \\"react-native\\";
declare type A = ReactNative.Constructor<number>;
"
`;
